src/main/java/teammates/ui/controller/InstructorFeedbackAbstractAction.java

```diff
@@ -100,21 +104,20 @@ protected FeedbackSessionAttributes extractFeedbackSessionData(boolean isCreatin
         type = getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_SESSIONVISIBLEBUTTON);
         switch (type) {
         case Const.INSTRUCTOR_FEEDBACK_SESSION_VISIBLE_TIME_CUSTOM:
-            attributes.setSessionVisibleFromTime(TimeHelper.combineDateTime(
-                        getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_VISIBLEDATE),
-                        getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_VISIBLETIME)));
+            Date visibleTimeLocal = TimeHelper.combineDateTime(
+                    getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_VISIBLEDATE),
+                    getRequestParamValue(Const.ParamsNames.FEEDBACK_SESSION_VISIBLETIME));
+            attributes.setSessionVisibleFromTimeUtc(
+                    TimeHelper.convertLocalDateToUtc(visibleTimeLocal, attributes.getTimeZone()));
             break;
         case Const.INSTRUCTOR_FEEDBACK_SESSION_VISIBLE_TIME_ATOPEN:
-            attributes.setSessionVisibleFromTime(Const.TIME_REPRESENTS_FOLLOW_OPENING);
+            attributes.setSessionVisibleFromTimeUtc(Const.TIME_REPRESENTS_FOLLOW_OPENING);
             break;
         case Const.INSTRUCTOR_FEEDBACK_SESSION_VISIBLE_TIME_NEVER:
-            attributes.setSessionVisibleFromTime(Const.TIME_REPRESENTS_NEVER);
+            attributes.setSessionVisibleFromTimeUtc(Const.TIME_REPRESENTS_NEVER);
             // Overwrite if private
-            attributes.setResultsVisibleFromTime(Const.TIME_REPRESENTS_NEVER);
+            attributes.setResultsVisibleFromTimeUtc(Const.TIME_REPRESENTS_NEVER);
             attributes.setFeedbackSessionType(FeedbackSessionType.PRIVATE);
-            if (!isCreatingNewSession) {
-                attributes.setEndTime(null);
```

Comment:
I suppose this is because the end time is null to begin with?

Reply:
Previously, it was set to null and would thus inherit the old value later on in `FeedbackSessionsLogic#updateFeedbackSession` (though I'm not sure this was actually intended):
https://github.com/TEAMMATES/teammates/blob/c18c8b3e762558e590d6997e3bde098968ae7cdc/src/main/java/teammates/logic/core/FeedbackSessionsLogic.java#L1159-L1161

One of the test cases updates the time zone and sets session visible time to never. The test was written in a way such that the local end time is not expected to change. Now that local end time interpretation depends on time zone, for the test to pass, the UTC end time needs to be updated so that the local end time remains the same.